### PR TITLE
Add function code and closure bytes into cache key

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

### CODE REVIEW

**Comment by stuartarchibald** [comment]
Opened for CI, see if this works. Trying to cache with a unique but reproducible key for closures.

File: numba/core/caching.py [hunk 1]
```diff
@@ -697,7 +699,15 @@ def _index_key(self, sig, codegen):
         Compute index key for the given signature and codegen.
         It includes a description of the OS and target architecture.
         """
-        return (sig, codegen.magic_tuple())
+        codebytes = self._py_func.__code__.co_code
+        if self._py_func.__closure__ is not None:
+            cvars = tuple([x.cell_contents for x in self._py_func.__closure__])
+        else:
+            cvars = b''
```
Comment (sklam) [inline numba/core/caching.py:706]: why changing the type of `cvars`?

Comment (sklam) [reply numba/core/caching.py:706]: is that intended for `cvarbytes` instead?

Comment (stuartarchibald) [reply numba/core/caching.py:706]: fixed.

**Pull request merged**